src/components/sidebar/GranulesTable.tsx

```diff
@@ -94,12 +95,44 @@ const GranuleTable = (props: GranuleTableProps) => {
     }
     return validInput
   }
+  
+  const alreadyAddedCyclePassScene = (cycle: string, pass: string, scene: string): boolean => {
+    let existingValue = false
+    addedProducts.forEach(product => {
+      if (product.cycle === cycle && product.pass === pass && product.scene === scene) {
+        existingValue = true
+      }
+    })
```

Comment:
I believe this can be done more efficiently with a map rather than iterating over the entire list every time a new item is added.

Reply:
Does a map not iterate over all the addedProducts the same amount of times a forEach loop does?

Reply:
This check is only here until I have can use the endpoint to check what cycle/scene/pass combinations are available